internal/caddy/config.go

```diff
@@ -22,58 +21,47 @@ type Storage struct {
 type Config struct {
 	Storage Storage                `json:"storage"`
 	Apps    map[string]interface{} `json:"apps"`
+	Logging caddy.Logging          `json:"logging"`
 }
 
 // ControllerConfig represents ingress controller config received through cli arguments.
 type ControllerConfig struct {
-	Email          string
-	AutomaticTLS   bool
-	TLSUseStaging  bool
 	WatchNamespace string
+	ConfigMapName  string
 }
 
-// NewConfig returns a plain slate caddy2 config file.
-func NewConfig(namespace string, cfg ControllerConfig) *Config {
-	acmeIssuer := caddytls.ACMEIssuer{
-		CA:    getCAEndpoint(cfg.TLSUseStaging),
-		Email: cfg.Email}
+// NewConfig returns a base plain slate caddy2 config file.
+func NewConfig(namespace string, cfgMapConfig *Config) *Config {
```

Comment:
I suppose we need to the `*Config` using the `ControllerConfig` here, something like:
```
func NewConfig(namespace string, controllerConfig * ControllerConfig) *Config { }
```
and handle all the required logics to generate the config in a single point, what's your opinion?

Reply:
IIUC, you propose to only use the `ControllerConfig` to generate a base Caddy config? 
It would mean that it would simply create a `*Config`, set the `Storage` provider and return. Then in `regenerateConfig()`, we check if `cfgMapConfig == nil` and:
- If `true`, we keep the Config provided at `/etc/caddy/config.json`
- Else, we create a base config

Am I in the right direction?